lib/test-collection.js

```diff
@@ -6,158 +13,173 @@ function TestCollection() {
 		throw new TypeError('Class constructor TestCollection cannot be invoked without \'new\'');
 	}
 
-	this.serial = [];
-	this.concurrent = [];
+	EventEmitter.call(this);
+
+	this.hasExclusive = false;
 	this.tests = {
+		concurrent: [],
+		serial: []
+	};
+
+	this.hooks = {
 		before: [],
 		beforeEach: [],
 		after: [],
 		afterEach: []
 	};
-	this.hasExclusive = false;
+
+	this._emitTestResult = this._emitTestResult.bind(this);
 }
 
+util.inherits(TestCollection, EventEmitter);
+
 TestCollection.prototype.add = function (test) {
 	var metadata = test.metadata;
 	var type = metadata.type;
 
 	if (!type) {
-		throw new Error('test type must be specified');
+		throw new Error('Test type must be specified');
+	}
+
+	if (!test.title && test.fn) {
+		test.title = fnName(test.fn);
+	}
+
+	if (test.title === 'callee$0$0') {
```

Comment:
Perhaps restore the `// workaround for Babel giving anonymous functions a name` comment. This looks like a magical string otherwise.